### PR TITLE
task: add OpenApi spec to metrics route

### PR DESCRIPTION
**Pull request by chriswk**
## About the changes
This adds openapi middleware for non-deprecated routes in src/lib/admin/metrics.ts

##
I created the schema for application from the joi schema we already had in place, I think I interpreted the strategies list correct.

### CODE REVIEW

File: src/lib/routes/admin-api/metrics.ts [hunk 1]
```diff
@@ -28,19 +31,68 @@ class MetricsController extends Controller {
...
-        this.get('/applications/:appName', this.getApplication);
+        this.route({
+            method: 'post',
+            path: '/applications/:appName',
+            handler: this.createApplication,
+            permission: UPDATE_APPLICATION,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'metricsCreateApplication',
```
Comment (olav) [inline src/lib/routes/admin-api/metrics.ts:42]: Do we need the `metrics` prefix in the operationIds? Maybe we should add more tags instead.

File: src/lib/routes/admin-api/metrics.ts [hunk 2]
```diff
@@ -28,19 +31,68 @@ class MetricsController extends Controller {
...
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'metricsGetApplication',
+                    responses: {
+                        200: createResponseSchema('applicationSchema'),
+                    },
+                }),
+            ],
+        });
```
Comment (olav) [inline src/lib/routes/admin-api/metrics.ts:95]: We should use the generated request/response types in the handlers as well.

**Review by olav — CHANGES_REQUESTED** [review]

File: src/lib/routes/admin-api/metrics.ts (hunk 1 above)
Comment (chriswk) [reply src/lib/routes/admin-api/metrics.ts:42]: Hmm, good question, I guess createApplication is only done in metrics so I can change the opId

File: src/lib/routes/admin-api/metrics.ts (hunk 2 above)
Comment (chriswk) [reply src/lib/routes/admin-api/metrics.ts:95]: Yes, will update

File: src/lib/routes/admin-api/metrics.ts (hunk 1 above)
Comment (chriswk) [reply src/lib/routes/admin-api/metrics.ts]: Done

File: src/lib/routes/admin-api/metrics.ts (hunk 2 above)
Comment (chriswk) [reply src/lib/routes/admin-api/metrics.ts:97]: Done

File: src/lib/routes/admin-api/metrics.ts [hunk 3]
```diff
@@ -28,19 +33,68 @@ class MetricsController extends Controller {
...
+        this.route({
+            method: 'delete',
+            path: '/applications/:appName',
+            handler: this.deleteApplication,
+            permission: UPDATE_APPLICATION,
+            acceptAnyContentType: true,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'metricsDeleteApplication',
```
Comment (olav) [inline src/lib/routes/admin-api/metrics.ts:61]: Could remove `metrics` here as well maybe, to match `createApplication`.

File: src/lib/routes/admin-api/metrics.ts [hunk 4]
```diff
@@ -28,19 +33,68 @@ class MetricsController extends Controller {
...
+        });
+        this.route({
+            method: 'get',
+            path: '/applications',
+            handler: this.getApplications,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'metricsGetApplications',
```
Comment (olav) [inline src/lib/routes/admin-api/metrics.ts:76]: Same here.

File: src/lib/routes/admin-api/metrics.ts [hunk 5]
```diff
@@ -28,19 +33,68 @@ class MetricsController extends Controller {
...
+        });
+        this.route({
+            method: 'get',
+            path: '/applications/:appName',
+            handler: this.getApplication,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'metricsGetApplication',
```
Comment (olav) [inline src/lib/routes/admin-api/metrics.ts:91]: Same here.

**Review by olav — APPROVED** [review]

**Pull request merged**